src/components/vm/filesystems/vmFilesystemsCard.jsx

```diff
@@ -146,7 +145,7 @@ const VmFilesystemAddModal = ({ connectionName, memory, memoryBacking, objPath,
         if (Object.getOwnPropertyNames(validationFailed).length == 0) {
             domainSetMemoryBacking({
                 connectionName, objPath,
-                type: memoryBackingType,
+                type: "memfd",
```

Comment:
Will that memfd contain the actual file system, or just some metadata for communciation? It seems fine for the latter, but squeezing possibly giga- or terabytes into a memfd is an utter abuse (and may not even work). If this contains the file system, what happens if you try to create a non-trivial one? Do our tests try to create a memfd-backed FS which exceeds the RAM size?

Reply:
The shared memory is used only for metadata according to https://lwn.net/Articles/774495/

Quoting the file is you 're not interested in all the doc:
```

- For data access inside guest, mmap portion of file in QEMU address
  space and guest accesses this memory using dax. That way guest page
  cache is bypassed and there is only one copy of data (on host). This
  will also enable mmap(MAP_SHARED) between guests.

- For metadata coherency, there is a shared memory region which contains
  version number associated with metadata and any guest changing metadata
  updates version number and other guests refresh metadata on next
  access. This is still experimental and implementation is not complete.
```

Does this cover you @martinpitt ? honestly I don't know the implementation details of this technology, but I read the libvirt/virtiofs docs and they did not specify any memory constraints of the host when using memfs for shared memory.

Reply:
Thanks! That sounds fine then, memfd is a rather nice technology for such stuff (and similar to classic shared memory, anyway).